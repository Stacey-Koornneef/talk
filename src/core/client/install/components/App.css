/* Here we add global stylings for body and document */
:global {
  body {
    margin: 0;

    /* Support for all WebKit browsers. */
    -webkit-font-smoothing: antialiased;
    /* Support for Firefox. */
    -moz-osx-font-smoothing: grayscale;
    background-color: #f2f2f2;
    color: #3b4a53;
  }
}

.root {
  width: 100%;
  box-sizing: border-box;
}

.container {
  padding: var(--spacing-unit) var(--spacing-unit) calc(2 * var(--spacing-unit))
    var(--spacing-unit);
  width: 1080px;
  margin: 0 auto;
}
