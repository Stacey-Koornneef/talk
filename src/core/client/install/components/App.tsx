import React, { Component } from "react";
import * as styles from "./App.css";

import { FormData } from "../containers/AppContainer";
import FinalStep from "../steps/components/FinalStep";
import InitialStep from "../steps/components/InitialStep";
import AddOrganizationContainer from "../steps/containers/AddOrganizationContainer";
import CreateYourAccountContainer from "../steps/containers/CreateYourAccountContainer";
import PermittedDomainsContainer from "../steps/containers/PermittedDomainsContainer";
import MainBar from "./MainBar";
import Wizard from "./Wizard";

export interface AppProps {
  currentStep: number;
  goToNextStep?: () => void;
  goToPreviousStep?: () => void;
  goToStep: (step: number) => void;
  saveData: (newData: {}) => void;
  data: Partial<FormData>;
}

class App extends Component<AppProps> {
  public render() {
    const {
      currentStep,
      goToNextStep,
      goToPreviousStep,
      goToStep,
    } = this.props;
    return (
      <div className={styles.root}>
        <MainBar />
        <div className={styles.container}>
          <Wizard
            currentStep={currentStep}
            goToNextStep={goToNextStep}
            goToPreviousStep={goToPreviousStep}
            goToStep={goToStep}
            className={styles.panes}
          >
            <InitialStep />
            <CreateYourAccountContainer
              data={this.props.data}
              saveData={this.props.saveData}
            />
            <AddOrganizationContainer
              data={this.props.data}
              saveData={this.props.saveData}
            />
            <PermittedDomainsContainer
              data={this.props.data}
              saveData={this.props.saveData}
            />
            <FinalStep />
          </Wizard>
        </div>
      </div>
    );
  }
}

export default App;
