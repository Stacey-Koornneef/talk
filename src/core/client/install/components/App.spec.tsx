import { shallow } from "enzyme";
import React from "react";

import { PropTypesOf } from "talk-framework/types";

import App from "./App";

it("renders sign in", () => {
  const props: PropTypesOf<typeof App> = {
    data: {
      organizationName: "",
      organizationContactEmail: "",
      organizationURL: "",
      email: "",
      username: "",
      password: "",
      confirmPassword: "",
      domains: [],
    },
    onSaveData: (newData: {}) => new Promise(resolve => resolve(props.data)),
  };
  const wrapper = shallow(<App {...props} />);
  expect(wrapper).toMatchSnapshot();
});
