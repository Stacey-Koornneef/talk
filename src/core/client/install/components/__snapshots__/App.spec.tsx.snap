// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sign in 1`] = `
<div
  className="App-root"
>
  <MainBar />
  <div
    className="App-container"
  >
    <WizardContainer
      data={
        Object {
          "confirmPassword": "",
          "domains": Array [],
          "email": "",
          "organizationContactEmail": "",
          "organizationName": "",
          "organizationURL": "",
          "password": "",
          "username": "",
        }
      }
      onSaveData={[Function]}
    />
  </div>
</div>
`;
